### DISCUSSION TITLE
Can I Use the Client of a Later Version to Connect to the Server of an Earlier Version?

### DISCUSSION THREAD

**Discussion by ouyang-xtt** [apache/pulsar#22259] [Q&A] [answered]
The client version is 3.0.x, and the server version is 2.10.4.

**Comment by jak78** [comment, marked as answer]
Hi @ouyang-xtt! That should work. Quoting the [documentation](https://pulsar.apache.org/docs/3.2.x/client-libraries/):
> A design goal of Pulsar is to ensure full compatibility between all versions of the client and the broker. When a client connects to a broker they agree upon a version of the protocol to use. As a result, new features that rely on an updates to the protocol are only available when using both newer clients and newer brokers.

It's important to note that version 2.10.4 is outdated and contains security vulnerabilities. We recommend upgrading to either a Long-Term Support (LTS) version (3.0.x) or a feature version (3.2.x). If migration to the 3.x series isn't feasible in short term, prioritize upgrading to version 2.10.6 as soon as possible.

Best,
